src/util/misc.ts

```diff
@@ -126,22 +127,11 @@ export function getOrInit<K, V>(key: K, m: Map<K, V>, def?: V): V {
 }
 
 export function getScopeUnit(
-    node: ContractDefinition | FunctionDefinition | VariableDeclaration
+    node: ContractDefinition | FunctionDefinition | VariableDeclaration | Statement
```

Comment:
Should `node`'s type also be an `AnnotationTarget`? Also if previous implementation is larger does not mean that it was ineffective. `ASTNode.getClosestParentByType()` actually compiles a closure and does a lot of traversal.

Reply:
Fixed in 681b5e5 .